src/shogun/multiclass/ShareBoost.cpp

```diff
@@ -43,6 +43,7 @@ SGVector<int32_t> CShareBoost::get_activeset()
 
 bool CShareBoost::train_machine(CFeatures* data)
 {
+	int32_t num_features = dynamic_cast<CDenseFeatures<float64_t>*>(m_features)->get_num_features();
```

Comment:
Lol, sorry we are miscommunicating.
So here you have a `dynamic_cast<CDenseFeatures<float64_t>*>(m_features)`

and a few lines below you have `m_fea = dynamic_cast<CDenseFeatures<float64_t> *>(m_features)`
I am talking about only casting once, not about the int